File: daily_challenge_leetcode/2654.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minOperations = function (nums) {
    const n = nums.length
    let count1 = 0
    for (let i = 0; i < n; i++) {
        if (nums[i] === 1) count1++
    }
    if (count1 > 0) return n - count1
    const gcd = (a, b) => {
        while (b !== 0) {
            [a, b] = [b, a % b]
        }
        return a
    }
    let minOps = Number.MAX_SAFE_INTEGER
    for (let i = 0; i < n - 1; i++) {
        let currentGCD = nums[i]
        for (let j = i + 1; j < n; j++) {
            currentGCD = gcd(currentGCD, nums[j])
            if (currentGCD === 1) {
                minOps = Math.min(minOps, j - i)
                break
            }
        }
    }
    if (minOps === Number.MAX_SAFE_INTEGER) {
        return -1
    }
    return minOps + (n - 1)
};

File: daily_challenge_leetcode/2654.cpp

// gcd Tc, = log(max(element))
// Tc = O(n*n*log(max(element)))
// Sc = O(1)

class Solution {
public:
    int minOperations(vector<int>& nums) {
        int n = nums.size();
        int count1 = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] == 1)
                count1++;
        }
        if (count1 > 0) {
            return n - count1;
        }
        int minOps = INT_MAX;
        for (int i = 0; i < n - 1; i++) {
            int GCD = nums[i];
            for (int j = i + 1; j < n; j++) {
                GCD = gcd(GCD, nums[j]);
                if (GCD == 1) {
                    minOps = min(minOps, j - i);
                    break;
                }
            }
        }
        if(minOps == INT_MAX){
            return -1;
        }
        return minOps + (n-1);
    }
};